yt_dlp/extractor/banbye.py

```diff
@@ -0,0 +1,114 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+
+from .common import InfoExtractor
+from ..utils import unified_timestamp
+
+
+class BanByeBaseIE(InfoExtractor):
+    _API_BASE = 'https://api.banbye.com'
+    _CDN_BASE = 'https://cdn.banbye.com'
+
+    def _extract_playlist(self, playlist_id):
+        data = self._download_json('%s/playlists/%s' % (self._API_BASE, playlist_id), playlist_id)
+        return self.playlist_result([
+            self.url_result('https://banbye.com/watch/%s' % video_id, ie_key='BanBye')
+            for video_id in data['videoIds']], playlist_id, data['name'])
+
+
+class BanByeIE(BanByeBaseIE):
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:www\.)?banbye.com/
+                        (?:en/)?watch/
+                        (?P<id>[^/?\#&]+)/?
+                        \??(?:playlistId=(?P<playlist_id>[^/?\#&]+))?'''
```

Comment:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?banbye.com/(?:en/)?watch/(?P<id>\w+)'
```
there may be other queries in the URL. Better to take playlist_id from parse_url